apps/emqx_rule_engine/src/emqx_rule_funcs.erl

```diff
@@ -1185,16 +1185,31 @@ function_literal(Fun, Args) ->
     {invalid_func, {Fun, Args}}.
 
 mongo_date() ->
-    erlang:timestamp().
+    maybe_isodate_format(erlang:timestamp()).
 
 mongo_date(MillisecondsTimestamp) ->
-    convert_timestamp(MillisecondsTimestamp).
+    maybe_isodate_format(convert_timestamp(MillisecondsTimestamp)).
 
 mongo_date(Timestamp, Unit) ->
     InsertedTimeUnit = time_unit(Unit),
     ScaledEpoch = erlang:convert_time_unit(Timestamp, InsertedTimeUnit, millisecond),
     convert_timestamp(ScaledEpoch).
```

Comment:
Q: doesn't this one need to be adapted as well?

Reply:
We do need to convert this when displaying it in sql-test.

Reply:
Thanks, will fix of course!